packages/engine/paima-funnel/src/funnels/parallelEvm/funnel.ts

```diff
@@ -122,9 +126,15 @@ export class ParallelEvmFunnel extends BaseFunnel implements ChainFunnel {
         cachedState.lastBlock = blocks[blocks.length - 1].blockNumber;
       }
 
-      while ((await this.updateLatestBlock()) === latestBlock) {
-        // wait for blocks to be produced
-        await delay(500);
+      // If we didn't break because there are no blocks in the chain then we
+      // need to wait for blocks to be produced.  But otherwise there is not
+      // much point in updating the value, since we may be really far away from
+      // the tip in the sync itself, and this adds a whole round trip.
+      if (to === latestBlock) {
+        while ((await this.updateLatestBlock()) === latestBlock) {
+          // wait for blocks to be produced
+          await delay(500);
+        }
       }
```

Comment:
```suggestion
      // We reach this part of the code if after we fetch blocks we still aren't done syncing.
      // There are 2 cases for this:
      // 1. We're still far behind the tip and we're catching up
      // For case (1), there is no need to do a network request or delay
      if (to !== latestBlock) continue;
      // 2. We're at the tip and we're waiting for more parallel chains blocks to be made to finalize the block
      // For case (2), we try update our cached latest block number in case it's stale,
      // and delay if we are truly waiting for more blocks to be made
      while ((await this.updateLatestBlock()) === latestBlock) {
        // wait for blocks to be produced
        await delay(500);
      }
```

Reply:
If you accept this suggestion, you also need to handle the 2 comments that come right after my suggested change (Github wouldn't let me modify them)